python/tempo/tsdf.py

```diff
@@ -750,63 +750,67 @@ def asofJoin(
         left_df = self.df
         right_df = right_tsdf.df
 
-        spark = SparkSession.builder.getOrCreate()
-        left_bytes = self.__getBytesFromPlan(left_df, spark)
-        right_bytes = self.__getBytesFromPlan(right_df, spark)
-
-        # choose 30MB as the cutoff for the broadcast
-        bytes_threshold = 30 * 1024 * 1024
-        if sql_join_opt & (
-            (left_bytes < bytes_threshold) | (right_bytes < bytes_threshold)
-        ):
-            spark.conf.set("spark.databricks.optimizer.rangeJoin.binSize", 60)
-            partition_cols = right_tsdf.partitionCols
-            left_cols = list(set(left_df.columns).difference(set(self.partitionCols)))
-            right_cols = list(
-                set(right_df.columns).difference(set(right_tsdf.partitionCols))
-            )
+        # test if the broadcast join will be efficient
+        if sql_join_opt:
+            spark = SparkSession.builder.getOrCreate()
+            left_bytes = self.__getBytesFromPlan(left_df, spark)
+            right_bytes = self.__getBytesFromPlan(right_df, spark)
+
+            # choose 30MB as the cutoff for the broadcast
+            bytes_threshold = 30 * 1024 * 1024
+            if (left_bytes < bytes_threshold) or (right_bytes < bytes_threshold):
+                spark.conf.set("spark.databricks.optimizer.rangeJoin.binSize", 60)
+                partition_cols = right_tsdf.partitionCols
+                left_cols = list(
+                    set(left_df.columns).difference(set(self.partitionCols))
+                )
+                right_cols = list(
+                    set(right_df.columns).difference(set(right_tsdf.partitionCols))
+                )
```

Comment:
```suggestion
                left_cols = list({left_df.columns} - {self.partitionCols})
                right_cols = list({right_df.columns} - {right_tsdf.partitionCols})
```

While this PR is open... 😁  as a general question, do we find this syntax to be more readable for set differences? We do potentially lose some clarity but it's less verbose